pkg/table/column.go

```diff
@@ -536,6 +536,24 @@ func GetColOriginDefaultValueWithoutStrictSQLMode(ctx sessionctx.Context, col *m
 	})
 }
 
+// CheckNoDefaultValueForInsert checks if the column has no default value before insert data.
+// CheckNoDefaultValueForInsert extrats the check logic from getColDefaultValueFromNil,
+// since getColDefaultValueFromNil function is public path and both read/write and other places use it.
+// But CheckNoDefaultValueForInsert logic should only check before insert.
+func CheckNoDefaultValueForInsert(ctx expression.BuildContext, col *model.ColumnInfo) error {
```

Comment:
You can use `errctx.Context` as the input argument. Seems this function does not need to build any expression

Reply:
nice catch, done.